### PR TITLE
Using golangci-lint instead of individual linters

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

-    Adding golangci-lint to the tests

-   Fixing most of the linting issue raised by golangci-lint
    
    The old linter where run without build flags. This meant that any file
    behing a build flag was not linted. This is why we have so many issue
    when switching to golangci-lint.

### Additional Notes

golangci-lint is not run on windows for now. The tool can't find our C compiler and GOARCH in the build container. This PR is big enough and has enough value in itself. Windows will come soon.

### CODE REVIEW

**Review by ofek — DISMISSED** [review]

**Review by mfpierre — DISMISSED** [review]
💯

**Comment by bai** [comment]
Out of curiosity, is this still something that's on your roadmap?

**Comment by hush-hush** [comment]
@bai Yes it is, just waiting for reviews from other teams. I'm going to ping them.

File: .golangci.yml
Comment (ogaca-dd) [inline .golangci.yml]: I think it would be nice to use https://github.com/kisielk/errcheck.

You can exclude `log.Warnf` and `log.Errorf` by adding 

```
linters-settings:
  errcheck:
    ignore: fmt:.*,github.com/DataDog/datadog-agent/pkg/util/log:.*
```

See [Excluding functions](https://github.com/kisielk/errcheck#excluding-functions) for more information.

File: pkg/forwarder/forwarder.go
Comment (ogaca-dd) [inline pkg/forwarder/forwarder.go]: What is the purpose of this comment?

File: tasks/go.py
Comment (ogaca-dd) [inline tasks/go.py]: Should it be `golangci-lint` instead of `golangci_lint`

**Review by ogaca-dd — DISMISSED** [review]
LGTM.

File: tasks/go.py
Comment (bai) [reply tasks/go.py]: Nit: `invokation` -> `invocation`

Comment (bai) [reply tasks/go.py]: Correct me if I'm wrong, `golangci_lint` task refers to this Invoke task: https://github.com/DataDog/datadog-agent/pull/4538/files#diff-ff17943812ec2171c673c7b301d38bc8R21 that uses underscore.

File: pkg/forwarder/forwarder.go
Comment (hush-hush) [reply pkg/forwarder/forwarder.go]: `// nolint` disable linting issue for that line. We don't use this variable for now but we want to keep it for future evolution

File: tasks/go.py
Comment (hush-hush) [reply tasks/go.py]: Exactly `golangci_lint` is the invoke command

Comment (hush-hush) [reply tasks/go.py]: invoke replace `_` by `-`, good catch !

**Review by ogaca-dd — DISMISSED** [review]

**Review by albertvaka — DISMISSED** [review]

**Pull request merged**
